converter/internal/otelcolconvert/converter_tailsamplingprocessor.go

```diff
@@ -0,0 +1,241 @@
+package otelcolconvert
+
+import (
+	"fmt"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/processor/tail_sampling"
+	"github.com/grafana/agent/converter/diag"
+	"github.com/grafana/agent/converter/internal/common"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor"
+	"go.opentelemetry.io/collector/component"
+)
+
+func init() {
+	converters = append(converters, tailSamplingProcessorConverter{})
+}
+
+type tailSamplingProcessorConverter struct{}
+
+func (tailSamplingProcessorConverter) Factory() component.Factory {
+	return tailsamplingprocessor.NewFactory()
+}
+
+func (tailSamplingProcessorConverter) InputComponentName() string {
+	return "otelcol.processor.tail_sampling"
+}
+
+func (tailSamplingProcessorConverter) ConvertAndAppend(state *state, id component.InstanceID, cfg component.Config) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	label := state.FlowComponentLabel()
+
+	args := toTailSamplingProcessor(state, id, cfg.(*tailsamplingprocessor.Config))
+	block := common.NewBlockWithOverride([]string{"otelcol", "processor", "tail_sampling"}, label, args)
+
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Converted %s into %s", stringifyInstanceID(id), stringifyBlock(block)),
+	)
+
+	state.Body().AppendBlock(block)
+	return diags
+}
+
+func toTailSamplingProcessor(state *state, id component.InstanceID, cfg *tailsamplingprocessor.Config) *tail_sampling.Arguments {
+	var (
+		nextMetrics = state.Next(id, component.DataTypeMetrics)
+		nextTraces  = state.Next(id, component.DataTypeTraces)
+		nextLogs    = state.Next(id, component.DataTypeLogs)
+	)
+
+	testEncode := encodeMapstruct(cfg.PolicyCfgs[0])
+	spew.Dump(testEncode)
+
+	return &tail_sampling.Arguments{
+		PolicyCfgs:              toPolicyCfgs(cfg.PolicyCfgs, state),
+		DecisionWait:            cfg.DecisionWait,
+		NumTraces:               cfg.NumTraces,
+		ExpectedNewTracesPerSec: cfg.ExpectedNewTracesPerSec,
+		Output: &otelcol.ConsumerArguments{
+			Metrics: toTokenizedConsumers(nextMetrics),
+			Logs:    toTokenizedConsumers(nextLogs),
+			Traces:  toTokenizedConsumers(nextTraces),
+		},
+	}
+}
+
+func toPolicyCfgs(cfgs []tailsamplingprocessor.PolicyCfg, state *state) []tail_sampling.PolicyConfig {
+	var out []tail_sampling.PolicyConfig
+	for _, cfg := range cfgs {
+		out = append(out, tail_sampling.PolicyConfig{
+			SharedPolicyConfig: toSharedPolicyConfig(cfg),
+			CompositeConfig:    toCompositeConfig(cfg.CompositeCfg),
+			AndConfig:          toAndConfig(cfg.AndCfg),
+		})
+	}
+	return out
+}
+
+func toSharedPolicyConfig(cfg tailsamplingprocessor.PolicyCfg) tail_sampling.SharedPolicyConfig {
+	return tail_sampling.SharedPolicyConfig{
```

Comment:
This will look repeated down below but we are in fact getting the same fields on 3 different structs. I'm open to suggestion but this is probably not worth more complexity or effort to save a little code.